src/Ormolu/Utils/Extensions.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+module Ormolu.Utils.Extensions
+  ( Extension (..),
+    getExtensionsFromCabalFile,
+    findCabalFile,
+    getCabalExtensionDynOptions,
+  )
+where
+
+import Control.Exception
+import Control.Monad.IO.Class
+import qualified Data.ByteString as B
+import Data.List (find)
+import Data.Map.Lazy (Map)
+import qualified Data.Map.Lazy as M
+import Data.Maybe (maybeToList)
+import qualified Distribution.ModuleName as ModuleName
+import Distribution.PackageDescription
+import Distribution.PackageDescription.Parsec
+import Distribution.Utils.Path
+import Language.Haskell.Extension
+import Ormolu.Config
+import System.Directory
+import System.FilePath
+import System.IO.Error (isDoesNotExistError)
+
+getExtensionsFromCabalFile ::
+  MonadIO m =>
+  -- | Path to cabal file
+  FilePath ->
+  -- | Map from Haskell source file paths (without any extensions)
```

Comment:
Why should without extensions?

Reply:
In order to support e.g. `hs-boot` files.